src/watertap_contrib/seto/unit_models/surrogate/vagmd_surrogate.py

```diff
@@ -0,0 +1,1309 @@
+import idaes.core.util.scaling as iscale
+from idaes.core.solvers import get_solver
+import idaes.logger as idaeslog
+
+# Import Watertap packages
+from watertap.core.util.model_diagnostics.infeasible import *
+
+_log = idaeslog.getLogger(__name__)
+__author__ = "Zhuoran Zhang"
+
+from idaes.core.util.model_statistics import degrees_of_freedom
+from idaes.core.util.scaling import (
+    calculate_scaling_factors,
+    unscaled_variables_generator,
+    badly_scaled_var_generator,
+)
+
+
+@declare_process_block_class("VAGMDSurrogate")
+class VAGMDData(UnitModelBlockData):
+    """
+    Vacuumed Membrane distillation (air-gapped) - batch operation model
+    """
+
+    CONFIG = ConfigBlock()
+
+    CONFIG.declare(
+        "dynamic",
+        ConfigValue(
+            domain=In([False]),
+            default=False,
+            description="Dynamic model flag - must be True",
+            doc="""Indicates whether this model will be dynamic or not,
+    **default** = False. """,
+        ),
+    )
+    CONFIG.declare(
+        "has_holdup",
+        ConfigValue(
+            default=False,
+            domain=In([False]),
+            description="Holdup construction flag - must be True",
+            doc="""Indicates whether holdup terms should be constructed or not.
+    **default** - False. The filtration unit does not have defined volume, thus
+    this must be False.""",
+        ),
+    )
+    CONFIG.declare(
+        "property_package_seawater",
+        ConfigValue(
+            default=useDefault,
+            domain=is_physical_parameter_block,
+            description="Property package to use for control volume",
+            doc="""Property parameter object used to define property calculations,
+    **default** - useDefault.
+    **Valid values:** {
+    **useDefault** - use default package from parent model or flowsheet,
+    **PhysicalParameterObject** - a PhysicalParameterBlock object.}""",
+        ),
+    )
+    CONFIG.declare(
+        "property_package_water",
+        ConfigValue(
+            default=useDefault,
+            domain=is_physical_parameter_block,
+            description="Property package to use for control volume",
+            doc="""Property parameter object used to define property calculations,
+    **default** - useDefault.
+    **Valid values:** {
+    **useDefault** - use default package from parent model or flowsheet,
+    **PhysicalParameterObject** - a PhysicalParameterBlock object.}""",
+        ),
+    )
+    CONFIG.declare(
+        "property_package_args",
+        ConfigBlock(
+            implicit=True,
+            description="Arguments to use for constructing property packages",
+            doc="""A ConfigBlock with arguments to be passed to a property block(s)
+    and used when constructing these,
+    **default** - None.
+    **Valid values:** {
+    see property package for documentation.}""",
+        ),
+    )
+    CONFIG.declare(
+        "high_brine_salinity",
```

Comment:
This seems like an odd configuration argument to have. It seems like the user would need to predetermine whether their brine salinity would surpass 175.3 g/L at any point during a batch cycle. Wondering if there is another (better) way to do this. Maybe there isn't.